lib/hlsdump.js

```diff
@@ -50,31 +50,31 @@ var Hlsdump = function (parameters, callback) {
 
     Hlsdump.prototype._concatTsFiles = function (files) {
         var self = this,
-            exec = require('child_process').exec,
-            i,
-            params = '';
+            concatStream = fs.createWriteStream(self.filename);
+
+        function _concatFiles() {
+            if (files.length) {
+                var readStream = fs.createReadStream(files.shift());
+
+                readStream.pipe(concatStream, {end: false});
+
+                readStream.on('error', function(err) {
+                    self.emit('error', err);
+                });
 
-        for (i in files) {
-            if (files.hasOwnProperty(i)) {
-                params = params + files[i] + ' ';
+                readStream.on('end', function() {
+                    _concatFiles();
+                });
+            } else {
+                self.emit('done');
```

Comment:
This should be:
```js
                concatStream.end(function() {
                    self.emit('done');
                });
```

Otherwise `self.filename` will not be unlocked and the temporary files can't be deleted while the application is running.